### PR TITLE
Expose image relative directory name

### PR DESCRIPTION
**Pull request by benjaminbarbe**
Useful with custom image asset delegate

### CODE REVIEW

File: lottie/src/main/java/com/airbnb/lottie/LottieImageAsset.java [hunk 1]
```diff
@@ -25,7 +27,7 @@ private Factory() {
 
     static LottieImageAsset newInstance(JSONObject imageJson) {
       return new LottieImageAsset(imageJson.optInt("w"), imageJson.optInt("h"), imageJson.optString("id"),
-          imageJson.optString("p"));
+          imageJson.optString("p"), imageJson.optString("u"));
```
Comment (gpeal) [inline lottie/src/main/java/com/airbnb/lottie/LottieImageAsset.java]: What is this?

Comment (piotrek1543) [reply lottie/src/main/java/com/airbnb/lottie/LottieImageAsset.java]: @gpeal he added fifth argument (check line 16.) called `dirName` with value of `imageJson.optString("u")`.

**Comment by gpeal** [comment]
@piotrek1543 @benjaminbarbe I see that. What is the purpose of the dirName that couldn't be accomplished with slashes in the file name?
The PR title also indicates that this is useful for usage with a image asset delegate which is why I was surprised to see additional json parsing and no new public apis.

**Comment by benjaminbarbe** [comment]
@gpeal @piotrek1543 The delegate pass the LottieImageAsset wich now expose the directory name

```java
mLottieAnimationView.setImageAssetDelegate(new ImageAssetDelegate() {
    @Override
    public Bitmap fetchBitmap(LottieImageAsset asset) {
        try {
            FileInputStream is = new FileInputStream(animationFolderPath + File.separator + asset.getDirName() + asset.getFileName());
            BitmapFactory.Options opts = new BitmapFactory.Options();
            opts.inScaled = true;
            opts.inDensity = 160;
            return BitmapFactory.decodeStream(is, null, opts);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        return null;
    }
});
```

It's useful when we didn't know in advance the hierarchy of the animation (ie: we download a full zip from network).

```json
{
  "v": "4.12.2",
  "fr": 25,
  "ip": 141,
  "op": 202,
  "w": 2732,
  "h": 1536,
  "nm": "Name",
  "ddd": 0,
  "assets": [
    {
      "id": "image_0",
      "w": 551,
      "h": 224,
      "u": "images/",
      "p": "img_0.png"
    },
    {
      "id": "image_1",
      "w": 1878,
      "h": 728,
      "u": "images/",
      "p": "img_1.png"
    }
  ],
  ...
}
```

[voiture.zip](https://github.com/airbnb/lottie-android/files/1443076/voiture.zip)

**Comment by benjaminbarbe** [comment]
Hello @gpeal 
Do you have any update on this?

**Comment by gpeal** [comment]
@bodymovin, the "u" folder is always `images/` right?

@benjaminbarbe

**Comment by benjaminbarbe** [comment]
For now, I think so. But it can be useful for future compatibility?

https://github.com/bodymovin/bodymovin/issues/183

**Comment by bodymovin** [comment]
@gpeal yes, the u folder is always images/
for the web player, you can pass a property to override it with a custom path.

**Pull request merged**